src/lib/components/Swap/index.tsx

```diff
@@ -54,34 +54,36 @@ export default function Swap(props: SwapProps) {
   useSyncSwapDefaults(props)
   useSyncConvenienceFee(props)
 
-  const { active, account, chainId } = useActiveWeb3React()
+  const { active, account } = useActiveWeb3React()
   const [wrapper, setWrapper] = useState<HTMLDivElement | null>(null)
 
   const [displayTxHash, setDisplayTxHash] = useAtom(displayTxHashAtom)
   const pendingTxs = usePendingTransactions()
   const displayTx = getSwapTx(pendingTxs, displayTxHash)
 
   const tokenList = useTokenList()
-  const onSupportedNetwork = useMemo(() => chainId && ALL_SUPPORTED_CHAIN_IDS.includes(chainId), [chainId])
+  const onSupportedNetwork = useOnSupportedNetwork()
   const isSwapSupported = useMemo(
     () => Boolean(active && onSupportedNetwork && tokenList?.length),
     [active, onSupportedNetwork, tokenList?.length]
   )
 
   const focused = useHasFocus(wrapper)
 
+  const interactionsDisabled = useMemo(() => !active || !onSupportedNetwork, [active, onSupportedNetwork])
```

Comment:
as simple booleans, doesn't need to use `useMemo`.
also, prefer a positive variable name, and I prefer to prefix booleans with "is" or "should" - `isInteractive`?